components/brave_shields/browser/ad_block_service.cc

```diff
@@ -78,4 +95,39 @@ void RegisterPrefsForAdBlockService(PrefRegistrySimple* registry) {
   registry->RegisterBooleanPref(kAdBlockCheckedDefaultRegion, false);
 }
 
+AdBlockPrefService::AdBlockPrefService(PrefService* prefs) : prefs_(prefs) {
+  pref_change_registrar_.reset(new PrefChangeRegistrar());
+  pref_change_registrar_->Init(prefs_);
+  pref_change_registrar_->Add(
```

Comment:
these are all currently showing warnings for pref observer found at shutdown in the network delegate. That might be because it's on the IO thread right now, but possibly needs `pref_change_registrar_.RemoveAll();` in `Shutdown`?

Reply:
Moving prefs here actually fixes these warnings. They existed, because network delegate outlives Profile and PrefService. Now the registrar correctly dies in time together with BCKS